### DISCUSSION TITLE
Hide the cmd window after starting

### DISCUSSION THREAD

**Discussion by laurin** [tauri-apps/tauri#4179] [Q&A] [answered]
On Windows, when I start the compiled `.exe` file or the application installed by the installer, a cmd window is opened that stays open for the whole lifetime of my app.  
Is there some clean way to quickly close this window or make it not show up in the first place?  
Of course I could try to find the window via windows user32 functions and then hide it, but that wouldn't be very clean.

**Comment by FabianLars** [comment, marked as answer]
This is what
```
#![cfg_attr(
    all(not(debug_assertions), target_os = "windows"),
    windows_subsystem = "windows"
)]
```
at the top of the generated main.rs is for.

**Reply by laurin** [reply]
Ah, thanks a lot, that works! 👌🏻  
Seems like I missed it somehow. That should probably be documented better, but at least it can now be found in this discussion.

**Reply by sujeetkalaskar** [reply]
Hi @FabianLars and Hi @laurin ,

I am also facing the same issue, where console windows is opened when complied .exe is started.

Could you please guide how to resolve the issue?

tried removing the below code, still, console window is opened
#![cfg_attr(
    all(not(debug_assertions), target_os = "windows"),
    windows_subsystem = "windows"
)]

**Reply by FabianLars** [reply]
@sujeetkalaskar You need this code at the top of your main.rs file to remove the console. Keep in mind that this only has an effect on `tauri build`, **not** on `tauri dev` and `tauri build --debug`

**Reply by sujeetkalaskar** [reply]
Hi @FabianLars ,

Thanks for the answer 👍 
I tried the approach, and it worked as you told me.
I was executing the process command to start the service, but I was unable to start it without opening the console window

**Reply by fal3n-4ngel** [reply]
Tried it but still consoles popping up whenever I run some docker cmd's

**Reply by FabianLars** [reply]
This either depends on how you spawn it, or it's just docker wanting to spawn a cmd window no matter what.

The first option is only relevant if you're on Windows and start docker in Rust, not using Tauri's Command api. Then you're likely missing the appropiate creation flag https://doc.rust-lang.org/std/process/struct.Command.html#method.creation_flags (`CREATE_NO_WINDOW`)

For the latter i can't think of anything to prevent it then.

**Reply by tempiltin** [reply]
![{22245CF6-462B-417F-BF6E-0C9E1B28C169}](https://github.com/user-attachments/assets/116d9603-d8fe-4e92-aa74-71fa48b38270)

hi i also have this problem cmd.exe window keeps popping up.
i already put this code at the top of main.rs file but the problem was not solved. i wrote special codes to make cmd window in all rust files work in hidden mode but the problem was not solved. i also used this:
const CREATE_NO_WINDOW: u32 = 0x08000000;

**Reply by FabianLars** [reply]
I can only think of 3 reasons:
1) you didn't apply it to main.rs correctly / or to the wrong file
2) you missed CREATE_NO_WINDOW on some commands 
3) You didn't properly rebuild your app yet or are testing the wrong one (it must be the one in target/release **not** target/debug)
